### DISCUSSION TITLE
Disable exposing metadata about SQL query string when exporting snapshot

### DISCUSSION THREAD

**Discussion by tbotalla** [grafana/grafana#38157] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
When exporting a snapshot of a dashboard that includes several panels, those panels have available the inspect option. That option permits seeing the panel json, and there is exposed the whole sql query string (snapshotData[0].meta.executedQueryString field) used for generating that panel. It would be great that the strip sensitive data option when generating a snapshot of a dashboard also includes this field.
This was tested using an SQL Server datasource on Grafana v7.1.1.

**Why is this needed**:
The fact that I want to share a snapshot and not the entire dashboard is that I want to see only the data queried, not how it is queried. Is it possible to generate the dashboard without exposing that information? 
Currently I am trying to share a dashboard on my site but exposing this kind of data is unacceptable.

**Comment by aocenas** [comment]
@torkelo @dprokop this seems like we should not even show the inspect panel in snapshots. Is that by design or did we just forgot to hide it there?

**Comment by torkelo** [comment]
No, inspect should be possible still. Allows you to view raw data and export to csv , and that should still be possible in a snapshot

**Comment by tbotalla** [comment]
@torkelo @aocenas I understand that you need to keep the inspect for that, but how would you do to embed on a website a panel or a snapshot without exposing that data? Is it possible? Maybe there is a feature that I am missing.
Thanks for answering.

**Comment by torkelo** [comment]
But yes, removing this meta data field should be part of the snapshot scrubbing
https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx#L114

**Comment by Imoustak** [comment]
Hello @torkelo are there any news regarding this ? Is it possible to achieve at the moment?

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
